source/fundamentals/crud/bulk.txt

```diff
@@ -0,0 +1,315 @@
+===============
+Bulk Operations
+===============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to use **bulk operations** in the
+MongoDB Java Driver. 
+
+To perform a create, replace, update or delete (CRUD) operation, you
+use their corresponding method. For example, to insert one document,
+update multiple documents, and delete one document in your collection,
+you use the ``insertOne()``, ``updateMany()`` and ``deleteOne()`` methods. 
+
+Instead of using three different methods to perform three tasks, you can
+use bulk operations, which uses one method to perform three tasks. Bulk
+operations consist of a large number of CRUD operations and can perform
+them using the ``bulkWrite()`` method. 
+
+Performing Bulk Operations
+--------------------------
+
+The ``bulkWrite()`` method accepts a list of bulk operations in the
+format of ``WriteModel`` documents. A ``WriteModel`` represents one of
+the CRUD operations. The method also optionally takes
+``BulkWriteOptions`` as a second parameter to specify the order you want
+the operations to be perform.
+
+The following snippet creates a ``List`` of bulk operations to pass into
+the ``bulkWrite()`` method: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/BulkWrite.java
+   :language: java
+   :dedent:
+   :start-after: begin bulkOpsList
+   :end-before: end bulkOpsList
+
+The following sections show how to create and use each type of
+``WriteModel`` document. The examples in each section contain the
+following documents in a collection:
```

Comment:
suggestion: Add a bit more introductory material for WriteModel. As is, it feels a bit abrupt.